libr/core/disasm.c

```diff
@@ -1887,6 +1887,13 @@ static void ds_show_functions(RDisasmState *ds) {
 			r_cons_printf ("%s  ", COLOR_RESET (ds));
 		}
 		r_cons_printf ("%d: ", r_anal_function_realsize (f));
+
+		// show function's realname in the signature if realnames are enabled 
+		if (core->flags->realnames) {
+			RFlagItem *flag = r_flag_get (core->flags, fcn_name);
```

Comment:
Why assume a flag by name instead of offset? If the function is renamed this wont work

Reply:
Aren't multiple flags possible for a single address? I see how it can become an issue

Reply:
Name is the correct approach. When a function is renamed, the flag is renamed in most cases as well. If there is a flag that has a different name, it is not the function's anymore. Also names uniquely identify flags while addresses don't.